packages/expo-ui/src/jetpack-compose/Button/index.tsx

```diff
@@ -85,6 +88,8 @@ export function transformButtonProps(props: ButtonProps): NativeButtonProps {
     text: children ?? '',
     systemImage,
     onButtonPressed: onPress,
+    // @ts-expect-error
+    modifiers: props.modifiers?.map((m) => m.__expo_shared_object_id__),
```

Comment:
```suggestion
    modifiers: createModifiers(props.modifiers),
```

maybe create a helper like `createModifiers` that to prevent exposing the shared object internal `__expo_shared_object_id__` and the `@ts-expect-error` everywhere

```js
function createModifiers(modifiers: ExpoModifiers[] | null) {
  // @ts-expect-error
  return modifiers?.map((m) => m.__expo_shared_object_id__),
}
```

Reply:
hmm, we can support any modifiers like that, but anytime you write a component that accepts SO as a prop you need this magic line. We should fix this in core at some point...

Reply:
alright since you have some idea to solve the problem in mind, then it makes sense to me. we can follow up later in a separate pr